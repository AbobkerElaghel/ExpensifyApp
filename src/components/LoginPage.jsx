import React from "react";
import {startGoogleLogin, startFacebookLogin} from "../actions/authAction";
import {connect} from "react-redux";
import GoogleButton from "./GoogleButton.jsx";
import FacebookButton from "./FacebookButton.jsx";

export const LoginPage = ({startGoogleLogin, startFacebookLogin}) => (
    <div className="box-layout">
        <div className="box-layout--box">
            <h1 className="box-layout--title">Expensify App</h1>
            <p>Keep track of what you spent with us.</p>
        <GoogleButton startGoogleLogin={startGoogleLogin}/>
        <FacebookButton startFacebookLogin={startFacebookLogin}/>
        </div>
    </div>
);

const mapDispatchToProps = dispatch => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);