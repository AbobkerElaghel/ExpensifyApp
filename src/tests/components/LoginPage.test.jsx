import React from "react";
import { shallow } from "enzyme";
import {LoginPage} from '../../components/LoginPage.jsx'

test("should render LoginPage component correctly", () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test("should call startLogin action", () => {
    const startLoginSpy = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLoginSpy}/>);
    wrapper.find('button').simulate('click');
    expect(startLoginSpy).toBeCalled();
});