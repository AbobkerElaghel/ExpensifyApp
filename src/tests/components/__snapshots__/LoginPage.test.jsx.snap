// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage component correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout--box"
  >
    <h1
      className="box-layout--title"
    >
      Expensify App
    </h1>
    <p>
      Keep track of what you spent with us.
    </p>
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
